src/test/java/org/openmainframeproject/cobolcheck/TestSuiteParserParsingTest.java

```diff
@@ -506,10 +506,146 @@ public void verify_can_attach_to_call_mock_with_arguments() {
+
+        List<String> expectedResult = new ArrayList<>();
+        expectedResult.add("           MOVE 1 TO UT-1-1-1-MOCK-EXPECTED");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-COUNT TO UT-ACTUAL-ACCESSES");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-EXPECTED TO UT-EXPECTED-ACCESSES");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-NAME TO UT-MOCK-OPERATION");
+        expectedResult.add("           SET UT-VERIFY-EXACT TO TRUE");
+        expectedResult.add("           ADD 1 TO UT-TEST-CASE-COUNT");
+        expectedResult.add("           PERFORM UT-ASSERT-ACCESSES");
+        testSuiteParser.getParsedTestSuiteLines(
+                new BufferedReader(new StringReader(testSuite.toString())),
+                numericFields);
+        List<String> actualResult = new ArrayList<>();
+        testSuiteParser.handleEndOfVerifyStatement(actualResult,Constants.EMPTY_STRING,Constants.EMPTY_STRING);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void verify_missing_call_mock_with_qualified_arguments() {
+        testSuite.append("       TESTSUITE \"call_mock_with_arguments\"");
+        testSuite.append("       TESTCASE \"qualified_arguments\"");
+        testSuite.append("       MOCK CALL 'PROG1' USING this in this-place, BY CONTENT other in this-place");
+        testSuite.append("          MOVE \"something\" TO this");
+        testSuite.append("          MOVE \"something else\" TO other");
+        testSuite.append("       END-MOCK");
+        testSuite.append("       PERFORM 000-START");
+        testSuite.append("       VERIFY CALL 'PROG1' USING");
+        testSuite.append("             this in new-place, BY CONTENT other in new-place");
+        testSuite.append("             HAPPENED ONCE");
+
+        List<String> expectedResult = new ArrayList<>();
+        expectedResult.add("           MOVE 1 TO UT-1-1-1-MOCK-EXPECTED");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-COUNT TO UT-ACTUAL-ACCESSES");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-EXPECTED TO UT-EXPECTED-ACCESSES");
+        expectedResult.add("           MOVE UT-1-1-1-MOCK-NAME TO UT-MOCK-OPERATION");
+        expectedResult.add("           SET UT-VERIFY-EXACT TO TRUE");
+        expectedResult.add("           ADD 1 TO UT-TEST-CASE-COUNT");
+        expectedResult.add("           PERFORM UT-ASSERT-ACCESSES");
+        try {
+            testSuiteParser.getParsedTestSuiteLines(
+                    new BufferedReader(new StringReader(testSuite.toString())),numericFields);
+            List<String> actualResult = new ArrayList<>();
+
+            testSuiteParser.handleEndOfVerifyStatement(actualResult, Constants.EMPTY_STRING, Constants.EMPTY_STRING);
+            // the following assert must not be performed!
+            fail("An exception should have been thrown :(");
+        } catch (VerifyReferencesNonexistentMockException mockException) {
```

Comment:
Junit has a built in function for testing if the right exception has been thrown: assertThrows(...). You can see the test  `it_throws_when_config_file_name_is_null()` for reference